### PR TITLE
test: add e2e tests for static examples

### LINKED ISSUE #414: E2E Static Examples
**User Story**

As a developer/Prism user I want to have control over how a response is generated (either with a static example or a dynamically generated payload).

**ACs**
- add test cases to test harness
- add at least the cases described in the examples section
- if any of the test fails then mark that it is an invalid result (e.g. mark as skipped) and report a Jira bug
- link that bug to this card 
- Make sure to add the actual tests to TestRails

**Additional resources**

https://stoplightio.atlassian.net/wiki/spaces/EN/pages/52887637/Prism#Prism-MockServerFeatures (please find relevant row)

**Example tests**

*NOTE: Prism2 doesn't have --dynamic CLI option**Running Prism with --dynamic only refers to Prism3*

JSON Schema Example
```
{
              type: 'object',
              properties: {
                pet: {
                  type: 'object',
                  properties: {
                    name: { type: 'string', examples: ['Clark'] },
                    middlename: { type: 'string', examples: ['J'], default: 'JJ' },
                  },
                },
              },
            }
```
**Example Tests**

When I send a request to an operation
And Prism was run with --dynamic
And this operation has an example
And I include ?__dynamic=false in the path
Then the static example should be returned in the payload

**Specific example**

Consider the following spec (example.oas3.yaml)
```
openapi: 3.0.2
paths:
  /todos:
    get:
      parameters:
        - name: title
          in: query
          style: form
          schema:
            enum:
              - eat
              - drink
      responses:
        200:
          description: Get Todo Items
          content:
            "application/json":
              schema:
                type: object
                properties:
                  title:
                    type: string
                    examples:
                      - Work
                      - Rest
                  description:
                    type: string
                  priority:
                    type: number
                    default: 0
```
Given 

prism mock example.oas3.yaml

When 

curl localhost:4010/todos

Expect output to be 

{"title":"Work","description":"string","priority":0}

Given 

prism mock basic.oas3.yaml --dynamic

When 

curl localhost:4010/todos?__dynamic=false

Expect 

{"title":"Work","description":"string","priority":0}

### PR DESCRIPTION
**Pull request by lag-of-death**
Add tests for static examples - closes #414

### CODE REVIEW

File: test-harness/specs/static_examples-AC-1.oas3.txt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+curl localhost:4010/todos
+
+Expect output to be
+
+{"title":"Work","description":"string","priority":0}
+====spec====
+info:
+  title: Sample API
+  version: 1.0.0
```
Comment (XVincentX) [inline test-harness/specs/static_examples-AC-1.oas3.txt]: You've been skipping this in the other documents, right? Maybe we can remove it from here too?

File: test-harness/specs/static_examples-AC-1.oas3.txt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+openapi: 3.0.2
+paths:
+  /todos:
+    get:
+      parameters:
```
Comment (XVincentX) [inline test-harness/specs/static_examples-AC-1.oas3.txt]: I think the parameter here is not required, we can remove it.

File: test-harness/specs/static_examples-AC-2.oas3.txt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+====spec====
+info:
+  title: Sample API
+  version: 1.0.0
+openapi: 3.0.2
+paths:
+  /todos:
+    get:
+      parameters:
+        - name: title
```
Comment (XVincentX) [inline test-harness/specs/static_examples-AC-2.oas3.txt]: Dtto, can we remove this part?

File: test-harness/specs/static_examples-AC-2.oas3.txt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+curl localhost:4010/todos?__dynamic=false
+
+Expect
+
+{"title":"Work","description":"string","priority":0}
+====spec====
+info:
+  title: Sample API
+  version: 1.0.0
```
Comment (XVincentX) [inline test-harness/specs/static_examples-AC-2.oas3.txt]: Yeah I think we can remove this

File: test-harness/specs/static_examples-AC-1.oas2.txt [hunk 5]
```diff
@@ -0,0 +1,42 @@
+====test====
+Given
```
Comment (philsturgeon) [inline test-harness/specs/static_examples-AC-1.oas2.txt]: I don't understand why we would do this? This seems to be duplicating information from the rest of the file, and I am not sure what benefit it serves? 

Yesterday we talked about simplifying the `==test==` content by making it more human, less gherkin, and avoiding duplication with the test file. This seems to have made it less human, more gherkin, increased duplication, and added some blank lines in for unknown reasons. 

> When prism is started with the --dynamic flag, responses should continue to match expected document structure

Done :)

**Comment by lag-of-death** [comment]
@XVincentX, I cleaned up the specs and also fixed some false positives. Turns out, that when there was no `-i` in `curl something`, then, the bodies from https://github.com/stoplightio/prism/blob/21ba780a85ef20a9313286dd511af18965ff591d/test-harness/index.ts#L62 were empty

**Comment by XVincentX** [comment]
@lag-of-death Ok that's the part I should check out in the harness, apparently when no -i the body is…gone. In meantime I've added the flag to all the missing tests so we tests the things :)

**Pull request merged**